EIPS/eip-4519.md

```diff
@@ -0,0 +1,237 @@
+---
+eip: 4519
+title: Smart Non Fungible Token (SmartNFT)
+description: A standard interface of smart non-fungible tokens representing smart assets (such as secure IoT devices) that can generate their own blockchain accounts and obey users.
+author: Javier Arcenegui <arcenegui@imse-cnm.csic.es>, Rosario Arjona <arjona@imse-cnm.csic.es>, Roberto Román <roman@imse-cnm.csic.es> and Iluminada Baturone <lumi@imse-cnm.csic.es>
+discussions-to: https://ethereum-magicians.org/t/new-proposal-of-smart-non-fungible-token/7677
+status: draft
+type: standards track
+category: erc
+created: 2021-12-03
+requires: 165, 721
+---
+
+## Abstract
+The novelty introduced in this EIP is the proposal of a Smart Non-Fungible Token (SmartNFT) to represent a physical smart asset, such as an IoT device. A SmartNFT is physically bound to a smart asset. A Smart Asset can hold an Ethereum address to participate in the Ethereum chain. It also serves as a utility to the user as it can establish secure communication with their owners and other Ethereum users. Likewise, it can operate dynamically with several operating modes associated with its token states. For example, an IoT device capable of generating its Ethereum Address from its private key and is also the sole owner of this private key. The IoT device does not store the private key but uses a Physical Unclonable Function (PUF) that allows recovering the private key. SmartNFTs extend ERC-721 non-fungible tokens (which only allow representing assets by a unique identifier, as a possession of an owner).
```

Comment:
```suggestion
The novelty introduced in this EIP is the proposal of a Smart Non-Fungible Token (SmartNFT) to represent a physical smart asset, such as an IoT device. A SmartNFT is physically bound to a smart asset. A Smart Asset can hold an Ethereum address to participate in the Ethereum chain. It also serves as a utility to the user as it can establish secure communication with their owners and other Ethereum users. Likewise, it can operate dynamically with several operating modes associated with its token states. For example, an IoT device capable of generating its Ethereum Address from its private key and is also the sole owner of this private key. The IoT device does not store the private key but uses a Physical Unclonable Function (PUF) that allows recovering the private key. SmartNFTs extend [ERC-721](./eip-721.md) non-fungible tokens (which only allow representing assets by a unique identifier, as a possession of an owner).
```